File: 100 problem/8sum_of_all_number.py


def sum_of_all_number(arr):
    sum = 0
    
    for i in arr:
        sum += i
    return sum

def sum_of_all_number_2(arr):
    sum = 0
    
    i = 0
    j = len(arr) -1
    
    while i < j:
        sum += arr[i] + arr[j]
        i += 1
        j -= 1
    
    if len(arr) % 2 != 0:
        sum += arr[len(arr)//2]
    return sum


arr = [10, 20, 30, 40, 50, -1, 20, 1, -1]


print(sum_of_all_number_2(arr))

File: 100 problem/8sum_of_all_number.js
/****************************************************************
 * WRITE A PROGRAM TO FIND THE SUM OF ALL ELEMENTS IN AN ARRAY. *
 ****************************************************************/



function sum_of_all_number(arr) {
    let sum = 0;
    let arr_size = arr.length;
    for(let i = 0; i < arr_size; ++i) sum += arr[i];

    return sum;
}




function sum_of_all_number_2(arr) {
    let sum = 0;
    let arr_size = arr.length;
    
    let i = 0, j = arr_size - 1;

    while(i < j) {
        sum += arr[i++] + arr[j--];
    }

    let mid = Math.floor(arr_size / 2);
    if(arr_size % 2 != 0) sum += arr[mid];

    return sum;
}


arr = [10, 20, 3, -2, 20, -21, 24, 100, -1]
console.log(sum_of_all_number_2(arr))